packages/client/src/interface/Login.ts

```diff
@@ -0,0 +1,41 @@
+import { z } from 'zod';
+import { IThunkApi } from './index';
+
+export interface ILoginProps {
+  login: string
+  password: string
+}
+
+export interface IErrorResponse {
+  reason: string
+}
+
+export interface IBaseApiProps {
+  thunkApi: IThunkApi
+}
+
+export interface ILoginApiProps extends IBaseApiProps {
+  data: ILoginProps
+}
+
+export type ILoginResponse = IErrorResponse | null
+
+export const UserModel = z.object({
+  id: z.number(),
+  first_name: z.string(),
+  second_name: z.string(),
+  display_name: z.string(),
+  login: z.string(),
+  email: z.string(),
+  phone: z.string(),
+  avatar: z.string(),
+});
+
+export type IUser = z.infer<typeof UserModel>
+
+export type IGetMeResponse = IUser | IErrorResponse
+
+export const LoginValidationModel = z.object({
+  login: z.string().min(4, 'Логин не может быть короче 4 символов'),
+  password: z.string().min(4, 'Пароль не может быть короче 4 символов'),
```

Comment:
вот тут дублирование неприятное. Много раз сталкивался с похожими ситуациями, когда приходил менеджер, говорит поправить 4 на 5. Ты идёшь и делаешь так

```diff
- .min(4, 'Логин не может быть короче 4 символов')
+ .min(5, 'Логин не может быть короче 4 символов')
```

забывая заменить текст сообщения, потому что всё впопыхах.

Тут лучше константу иметь и подставлять её в
```suggestion
  login: z.string().min(MIN_LOGIN, `Логин не может быть короче ${MIN_LOGIN} символов`),
  password: z.string().min(MIN_PASSWORD, `Пароль не может быть короче ${MIN_PASSWORD} символов`),
```

Reply:
спасибо, вынес)